src/seq/randseq.jl

```diff
@@ -0,0 +1,88 @@
+# Random Sequence Generator
+# =========================
+#
+# Random sequence generator.
+#
+# This file is a part of BioJulia.
+# License is MIT: https://github.com/BioJulia/Bio.jl/blob/master/LICENSE.md
+
+# abstract sequence generator type
+abstract SequenceGenerator{T}
+
+# Sequence generator of stationary distributions.
+immutable StationaryGenerator{T} <: SequenceGenerator{T}
+    elems::Vector{T}
+    probs::Vector{Float64}
+
+    function StationaryGenerator(elems, probs)
+        @assert sum(probs) ≤ 1
+        @assert length(elems) == length(probs) + 1
+        return new(copy(elems), copy(probs))
+    end
+end
+
+function StationaryGenerator{T,U<:AbstractFloat}(
+        elems::AbstractVector{T}, probs::AbstractVector{U})
+    return StationaryGenerator{T}(elems, probs)
+end
+
+alphatype(::Type{DNANucleotide}) = DNAAlphabet{4}
+alphatype(::Type{RNANucleotide}) = RNAAlphabet{4}
+alphatype(::Type{AminoAcid})     = AminoAcidAlphabet
+
+function randseq{T}(generator::StationaryGenerator{T}, len::Integer)
```

Comment:
I'm not sure about the demand of random sequence generator, so I thought it would be better to start it as a minimum viable utility. `SequenceGenerator` may be worth exporting so that users can define their own generators more easily.